### ISSUE TITLE
🐛 Accessibility | Screen reader announces old message when the menu is open

### ISSUE THREAD

**Issue by mauli0110** [GCTC-NTGC/gc-digital-talent#15103]
### 🐛 Bug and Expected Behaviour

On mobile view of the application, the screen reader announces 'Main menu navigation..' instead of 'Close menu button expanded' when the menu is expanded or open

Expected outcome: Screen reader should announced the 'Close menu button expanded' as expected

### 🕵️ Details

Environment: UAT
Mobile view

### 📋 Steps to Reproduce

1. Launch GC Digital talent
2. Open Dev tools and resize the viewport until the mobile menu appears
3. Open accessibility or screen reader, use tab key to navigate to 'Open menu' button
4. Click on Open Menu button
5. The screen reader announces 'Main Menu navigation ... '

### 📸 Screenshot

https://github.com/user-attachments/assets/7f5fc919-e018-4853-b284-4967fd8b2baf

### ✅ Acceptance Criteria

- [ ] Screen reader announces 'Main menu navigation...' when the menu is opened

### 🔄 Regression

No

**Comment by bella77-69** [comment]
@mauli0110 I checked this on UAT with NVDA this morning, and it announces it: 
- "Open menu button collapsed" when the menu button is closed
- "Close menu button expanded" when it is open

Not sure why it wasn't working on UAT on Friday?? 🤷‍♀️ but it seems to be working fine today.

**Comment by mauli0110** [comment]
> [@mauli0110](https://github.com/mauli0110) I checked this on UAT with NVDA this morning, and it announces it:
> 
> * "Open menu button collapsed" when the menu button is closed
> * "Close menu button expanded" when it is open
> 
> Not sure why it wasn't working on UAT on Friday?? 🤷‍♀️ but it seems to be working fine today.

Yeah sure, I will try this today and see if working. Thanks @bella77-69 😊

**Comment by brindasasi** [comment]
It announces when the button is on focus. 
When you click on the Open Menu , the focus goes to Home icon inside the menu and start announcing that. 
When the focus comes back to Close Menu it announces it. 

Its a bit diff behavior from the top groups like admin view , community view. 

But I guess this is what expected.

**Comment by esizer** [comment]
Okay, I have done some more testing and I was able to reproduce the issue. It was pretty hard to do but I could do it! I'm not sure what is leading to the issue itself since I cannot reliably reproduce it but, something very confusing is happening.

So, what happens is, sometimes when the menu opens, the focus shifts to the home button. This is actually intended based on the code we have! Though, it is very bugggy it seems 😅 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/9ebc89d04708bdd31f17ae5dae6911301e32871b/apps/web/src/components/NavMenu/Menu.tsx#L115-L117

Now, since our menu is non-modal, this might not be the best behaviour. 

I also can't navigate the menu on safari when voiceover is on 😭
